src/s2/rect.rs

```diff
@@ -340,7 +377,7 @@ fn intersects_lat_edge(a: &Point, b: &Point, lat: Angle, lng: interval::Interval
     // Extend this to an orthonormal frame (x,y,z) where x is the direction
     // where the great circle through AB achieves its maximium latitude.
     let y = z.cross(&Point::from_coords(0., 0., 1.)).normalize();
-    let x = y.cross(&z);
+    let x = y.cross(&z).normalize();
```

Comment:
For reference the C++ version expects a unit vector https://github.com/google/s2geometry/blob/98f435222f24cd3865619f2df9e02f630b8d5d53/src/s2/s2latlng_rect.cc#L401

I'm guessing the Go implementation normalizes it upstream